### PR TITLE
PLAT-1954 Tox support and requirements cleanup

### PR DESCRIPTION
**Pull request by jmbowman**
Start using pip-tools for better management of dependencies, and correctly support testing multiple versions of Django via tox.  For now, only Django 1.8 is tested on Travis and expected to pass.

Also upgraded from `pep8` to `pycodestyle`, and allowed the `requests` version to increase in order to avoid conflicts with the requirements of other installed packages.

### CODE REVIEW

File: requirements/travis.in [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Requirements for running tests in Travis
+
+coveralls                 # Code coverage reporting
+pep8                      # PEP 8 compliance validation
```
Comment (bmedx) [inline requirements/travis.in]: Is this still needed?

Comment (jmbowman) [reply requirements/travis.in]: Nope, good catch; removing.

File: tox.ini [hunk 2]
```diff
@@ -13,8 +17,19 @@ deps =
     django19: Django>=1.9,<1.10
     django110: Django>=1.10,<1.11
     django111: Django>=1.11,<2.0
-    -rrequirements.txt
+    -r{toxinidir}/requirements/test.txt
 passenv =
     TRAVIS*
+whitelist_externals =
+    rake
```
Comment (doctoryes) [inline tox.ini:24]: Do you plan to remove rake in your XQueue work? It'd be nice to get rid of it here - though probably not a priority. Maybe a ticket is in order if this work doesn't remove it?

**Review by doctoryes — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (jmbowman) [reply tox.ini:24]: There's already a ticket for that that Kevin plans to handle: [OPS-2716](https://openedx.atlassian.net/browse/OPS-2716).

**Pull request merged**